packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -109,11 +102,15 @@ class _ListenerEntry extends LinkedListEntry<_ListenerEntry> {
 ///
 ///  * [ValueNotifier], which is a [ChangeNotifier] that wraps a single value.
 class ChangeNotifier implements Listenable {
-  LinkedList<_ListenerEntry>? _listeners = LinkedList<_ListenerEntry>();
+  int _count = 0;
+  List<VoidCallback?> _listeners = List<VoidCallback?>.filled(0, null);
```

Comment:
Since the length is specified as 0, what does "filled" get us here over just `const <VoidCallback?>[]`?

Reply:
@mraleph would explain this better than me, but in my understanding `const <VoidCallback?>[]` would create a growable array and inlining wouldn't work (because we use fixed-length array everywhere else). So it would create a drop in the performances.

Reply:
To be more precise: `const []` would be an instance of `_ImmutableList` which is yet another private class implementing `List` (in addition to a fixed-length `_List` and `_GrowableList`). If one wants operations like `_listeners[]` and `_listeners.length` to be inlined then you need to ensure that AOT compiler can infer _concrete_ type for this field - meaning that should not mix instances of different type in it. 

You can actually optimize this a bit by doing:

```dart
static final List<VoidCallback?> _emptyListeners = List<VoidCallback?>.filled(0, null);
List<VoidCallback?> _listeners = _emptyListeners;
```

then you avoid allocation of an empty list for each new `ChangeNotifier`

Reply:
Interesting, thanks for the explanation. May be worthwhile to leave a comment in the code.